### DISCUSSION TITLE
How to decide value of MAX_ITER with respect to our custom dataset?

### DISCUSSION THREAD

**Discussion by dsbyprateekg** [facebookresearch/detectron2#3718] [General]
Hi,
Is there any formula to decide cfg.SOLVER.MAX_ITER value?
FOr example, I have total 1300 training +1300 validation images and I have given max_iter values as 55000. Is it right?
Please share your thoughts on it.

**Comment by HuangChiEn** [comment]
Depend on how many epoch you want to run to train your model.
Suppose, i want to train the model in 100 epoch with 32 batch size for your customized dataset. 
then,
```
    batch_size = 32    
    epoch = 100
    train_ds_size = 1300

    # you need to find cfg.SOLVER.MAX_ITER to satisfy this equation..
    epoch = cfg.SOLVER.MAX_ITER * batch_size / train_ds_size
```

yes, such **user un-friendly** design of detection is ugly, but it's easy to implement the optimizer. 
For example, LARS, Multi-step optimizer rely on the step information though..

good luck!
